test/e2e/agent/recipes_test.go

```diff
@@ -103,7 +105,7 @@ func TestFleetKubernetesIntegrationRecipe(t *testing.T) {
 			WithDefaultESValidation(agent.HasWorkingDataStream(agent.MetricsType, "elastic_agent.elastic_agent", "default")).
 			WithDefaultESValidation(agent.HasWorkingDataStream(agent.MetricsType, "elastic_agent.filebeat", "default")).
 			// TODO reenable when https://github.com/elastic/beats/issues/27127 is fixed
-			//WithDefaultESValidation(agent.HasWorkingDataStream(agent.MetricsType, "elastic_agent.fleet_server", "default")).
+			// WithDefaultESValidation(agent.HasWorkingDataStream(agent.MetricsType, "elastic_agent.fleet_server", "default")).
```

Comment:
This makes me realize that https://github.com/elastic/beats/issues/27127 is resolved and we can uncomment those lines. But let's keep that PR focused and maybe do it in a few days after we get back to a state where e2e tests pass again.